src/pages/index.js

```diff
@@ -68,37 +111,117 @@ profileEditButton.addEventListener("click", () => {
   jobInput.value = currentUser.job;
 
   profileModal.open();
-  profileFormValidator.resetValidation();
+  formValidators["edit-profile-form"].resetValidation();
 });
 
 addNewCardButton.addEventListener("click", () => {
+  // formValidators["add-card-form"].disableButton();
   cardModal.open();
 });
 
+profileAvatarButton.addEventListener("click", () => {
+  avatarModal.open();
+});
+
 //FUNCTIONS:
 
 function createCard(cardData) {
-  const newCard = new Card(cardData, "#card-template", () => {
-    imagePreviewModal.open(cardData);
-  });
+  const newCard = new Card(
+    cardData, //data
+    "#card-template", //cardSelector
+    () => {
+      //handleImageClick
+      imagePreviewModal.open(cardData);
+    },
+    handleDeleteSubmit,
+    handleLikeClick
+  );
+
   return newCard.generateCard();
 }
 
+function handleSubmit(request, modalInstance, loadingText = "Saving...") {
+  //change button text:
+  modalInstance.renderLoading(true, loadingText);
+  request()
+    .then(() => {
+      modalInstance.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      modalInstance.renderLoading(false);
+    });
+}
+
 function handleProfileFormSubmit(inputValues) {
-  console.log(inputValues.name);
-  userInfo.setUserInfo({
-    nameInput: inputValues.name,
-    jobInput: inputValues.job,
-  });
-  profileModal.close();
+  // we create a function that returns a promise
+  function makeRequest() {
+    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
+    return api
+      .updateUserInfo(inputValues.name, inputValues.job)
+      .then((userData) => {
+        userInfo.setUserInfo({
+          nameInput: inputValues.name,
+          jobInput: inputValues.job,
+        });
+      });
+  }
+  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
+  handleSubmit(makeRequest, profileModal);
 }
 
 function handleAddCardFormSubmit(inputValues) {
-  const name = inputValues.title;
-  const link = inputValues.url;
-  const cardData = { name, link };
+  function makeRequest() {
+    return api.addNewCard(inputValues.title, inputValues.url).then((res) => {
+      cardsSection.prependItem(createCard(res));
+      cardModal.resetFormAndDisableButton();
```

Comment:
Please, make only method `resetForm` inside `ModalWithForm` because disabling is handled by `FormValidator`


```js
  resetForm() {
    this.modalForm.reset();
  }

```


Now only class `FormValidator` should **`disable/enable`** submit buttons and remove errors from the inputs (if you do it)


So, here will be 2 lines of code


```js
      cardModal.resetForm();  // reset the form
      formValidators["add-card-form"].disableButton();  // disalbe the button

```